tests/token/conftest.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import annotations
+
+from typing import Dict
+
+import pytest
+from pydantic import validator
+from pydantic.dataclasses import dataclass
+from pydantic.main import BaseModel
+
+from spylib.token import (
+    OfflineTokenABC,
+    OfflineTokenResponse,
+    OnlineTokenABC,
+    OnlineTokenResponse,
+)
+
+test_database: Dict[str, Dict] = {
+    "online": {},
+    "offline": {},
+}
+
+
+# These are the fixtures for the subdirectories
+
+
+@pytest.fixture(scope='session', autouse=True)
+def database():
+    return test_database
+
+
+@pytest.fixture(scope='session', autouse=True)
+def OnlineToken():
+    global test_database
+    test_database = {"offline": {}, "online": {}}
+
+    class OnlineToken(OnlineTokenABC):
+        async def save_token(self):
+            test_database['online'][self.store_name] = {}
+            test_database['online'][self.store_name][self.associated_user.id] = self
+
+        @classmethod
+        async def load_token(cls, store_name: str, user_id: str) -> OnlineToken:
+            return test_database['online'][store_name][user_id]
+
+    yield OnlineToken
+
+    test_database = {"offline": {}, "online": {}}
+
+
+@pytest.fixture(scope='session', autouse=True)
+def OfflineToken():
+    global test_database
+    test_database = {"offline": {}, "online": {}}
+
+    class OfflineToken(OfflineTokenABC):
+        async def save_token(self):
+            test_database['offline'][self.store_name] = self
+
+        @classmethod
+        async def load_token(cls, store_name: str) -> OfflineToken:
+            return test_database['offline'][store_name]
+
+    yield OfflineToken
+
+    test_database = {"offline": {}, "online": {}}
+
+
+@pytest.fixture(scope='session', autouse=True)
+def online_token_data() -> OnlineTokenResponse:
+    return OnlineTokenResponse(
+        access_token='ONLINETOKEN',
+        scope=','.join(['write_products', 'read_customers']),
+        expires_in=86399,
+        associated_user_scope=','.join(['write_orders', 'read_products']),
```

Comment:
As you read `associated_user_scope` has to be a subset of `scope` so:
```suggestion
        scope=','.join(['write_products', 'read_customers']),
        expires_in=86399,
        associated_user_scope=','.join(['read_products']),
```

Reply:
Good point, I think this should probably be checked in some middleware at some point?

Reply:
Hummmmm If you don't mind making an issue for us to get back to it. I have learned that you can't always prevent the users from doing something stupid. So while we want to prevent the preventable, we shouldn't over do it either